### ISSUE TITLE
Typescript generations ignores enum

### ISSUE THREAD

**Issue by floriangoeldi** [grpc/grpc-web#487]
Hi there!

For a prototype I'm working with grpc-web and typescript. I ran there in the following error.
In my proto file I have defined this enum:

```
enum LogicalOperator
{
	And = 0;
	Or = 1;
}
```

which is then not generated in the typescript files, which results in a build error:

`error TS2304: Cannot find name 'LogicalOperator'`

To workaround this build error I am currently copy pasting my typescript export of this enum:

```
export enum LogicalOperator{
  And = 0,
  Or = 1
}
```

into my service_pb.d.ts files and then it builds correctly.

Is there a reason those enums are ignored therefore am I missing something here? :)

**Comment by stanley-cheung** [comment]
This is likely already fixed in `master` that's yet (close) to be released.

**Comment by floriangoeldi** [comment]
Oh that's nice! Thanks for your fast response.

Since I'm working on a windows machine I will await the next release and live with my manual labor. 🙈

**Comment by jonahbron** [comment]
@stanley-cheung Any idea when this will be released?

**Comment by jonahbron** [comment]
@stanley-cheung I'm trying to build the plugin myself in a Docker container based on your instructions, but it gives this error:

```
cd "/grpc-web"/javascript/net/grpc/web && make install
make[1]: Entering directory '/grpc-web/javascript/net/grpc/web'
g++ -std=c++11 -I/usr/local/include -pthread  -c -o grpc_generator.o grpc_generator.cc
grpc_generator.cc:19:53: fatal error: google/protobuf/compiler/code_generator.h: No such file or directory
 #include <google/protobuf/compiler/code_generator.h>
                                                     ^
compilation terminated.
<builtin>: recipe for target 'grpc_generator.o' failed
make[1]: Leaving directory '/grpc-web/javascript/net/grpc/web'
make[1]: *** [grpc_generator.o] Error 1
Makefile:94: recipe for target 'install-plugin' failed
make: *** [install-plugin] Error 2
```

It seems like there's a dependency missing, but I see nothing in the project on how to get it.  Here's an excerpt from my Dockerfile:
```
ARG PROTOC_GEN_GRPC_WEB_VERSION=ef94d2e7e0b866be9822eff22c56b089e64e870b
RUN git clone https://github.com/grpc/grpc-web.git && \
    cd grpc-web && \
    git checkout `${PROTOC_GEN_GRPC_WEB_VERSION}` && \
    make install-plugin
```

**Comment by stanley-cheung** [comment]
You need to install the protobuf as a library first. See how we did it in our docker images for test here: https://github.com/grpc/grpc-web/blob/master/net/grpc/gateway/docker/prereqs/Dockerfile#L36-L40

**Comment by jonahbron** [comment]
Lol, I tried to fix it by copying `github.com/protocolbuffers/protobuf/src` into `/usr/local/include`. 

Your suggestion fixed it, thank you.  Here's the entire Dockerfile portion that can compile/install protoc and protoc-gen-grpc-web.

```
ARG PROTOC_VERSION=3.7.0
ARG PROTOC_GEN_GRPC_WEB_VERSION=ef94d2e7e0b866be9822eff22c56b089e64e870b

# Packages required for build
RUN apt update && \
    apt install -y unzip make g++ autoconf libtool

# Download and install pre-compiled protoc
RUN cd /tmp && \
    wget -O protoc.zip https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.zip && \
    unzip protoc.zip bin/protoc -d . && \
    rm protoc.zip && \
    mv bin/protoc /usr/local/bin/ && \
    rm -rf bin/

# Compile/install protoc-gen-grpc-web from source
RUN git clone https://github.com/grpc/grpc-web.git && \
    cd grpc-web && \
    git checkout ${PROTOC_GEN_GRPC_WEB_VERSION} && \
    rm -rf third_party && \
    git checkout third_party && \
    ./scripts/init_submodules.sh && \
    cd third_party/grpc && \
    make install && \
    cd third_party/protobuf && \
    make install && \
    cd ../../../.. && \
    make install-plugin
```

**Comment by jonahbron** [comment]
Also I can confirm that running the complied version from `master` does include enums properly.  Thanks @stanley-cheung!

**Comment by stanley-cheung** [comment]
This should be now fixed with release `1.0.4`.
